src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala

```diff
@@ -0,0 +1,191 @@
+package io.iohk.ethereum.db.dataSource
+
+import java.util.concurrent.locks.ReentrantReadWriteLock
+
+import org.rocksdb._
+
+class RocksDbDataSource(private var db: RocksDB, private val rocksDbConfig: RocksDbConfig) extends DataSource {
+
+  /**
+    * This function obtains the associated value to a key, if there exists one.
+    *
+    * @param namespace which will be searched for the key.
+    * @param key       the key retrieve the value.
+    * @return the value associated with the passed key.
+    */
+  override def get(namespace: Namespace, key: Key): Option[Value] = {
+    RocksDbDataSource.dbLock.readLock().lock()
+    try {
+      val readOptions = new ReadOptions().setVerifyChecksums(rocksDbConfig.verifyChecksums)
```

Comment:
Maybe it is worth passing one ` ReadOptions` object to this `RocksDbDataSource`, it seems costly to create it before each read  ? (it can be created in `apply` or ` createDb` functions)